okio/src/test/java/okio/BufferTest.java

```diff
@@ -582,4 +583,14 @@
     Buffer buf = new Buffer().write(new byte[] { 0, 1, 2, 3 });
     assertEquals(1, buf.snapshot(1).size());
   }
+
+  @Test public void readByteArrayThrowsEofException() throws Exception {
```

Comment:
Can you try putting this in `BufferedSourceTest`? This will ensure behavior parity between `RealBufferedSource` and `Buffer` for this method.